### ISSUE TITLE
Add temporary holds along with permanent hold

### ISSUE THREAD

**Issue by psfales** [csm10495/pyhtcc#8]
I believe it would be fairly straightforward to implement 

set_temp_cool_setpoint()
set_temp_heat_setpoint()
end_hold()

(Or perhaps set_cool_setpoint() and set_heat_setpoint() would be better names)

Currently, set_permananent_heat_setpoint() contains tthis block:

```
 return self.submit_control_changes({
            'HeatSetpoint' : temp,
            'StatusHeat' : 2,
            'StatusCool' : 2,
            'SystemSwitch' : 1,
        })
        
```

Empirically, it seems that all that is needed for a temporary hold is to set StatusHeat to 1 instead of 2

You can use a StatusHeat value of 0, to end any (temporary or permanent) hold and start running the regular schedule

**Comment by csm10495** [comment]
Doesn't a temp hold also use a Heat/CoolNextPeriod to specify which 15-min marker the hold should last until? If not having that, i'm not sure when the tmp period would end.

**Comment by psfales** [comment]
Hmm, I haven't tried that test yet.  (I will do so)   But if the web site is to be believed, it works the same as when you change the temperature by using the physical buttons.   When I change the temperature using HeatSetpoint with a new temperature, and StatusHeat set to 1, the web site says "Hold Until 10:00 PM"   (where 10:00 is the next scheduled program change)

A subsequent call with StatusHeat set to 0 reverts to normal operation (same as pressing the "cancel" button on the display)

**Comment by csm10495** [comment]
I think that may also depend on if you have a schedule. I don't have a schedule, so it seems to just pick a time to popup as a temp hold. 

Maybe a function that has an optional time to go until (aligned to 15 minutes). If no time given.. it'll go until the thermostat wants it to stop the temp hold. 

Then there can still be a cancel hold function that just cancels it like you said.

**Comment by psfales** [comment]
I was going to try and test it without a schedule, but now that I have a schedule, there doesn't seem to be any way to remove it!

Your suggestion to have a temporary hold with an optional end time sounds fantastic, assuming that the thermostat is capable of doing that.

**Comment by csm10495** [comment]
Yeah i think the biggest limitation on the end time is it must be within the next 24 hours. Like you can't say 'temp until next week at noon'. It has to be temp until a specific time (15 min aligned) in the next 24 hours.

Are you interested in providing a PR?

**Comment by psfales** [comment]
I don't know enough about git to be able to provide a PR.  Sorry.

**Comment by csm10495** [comment]
No problem. I'll get to this eventually. Thanks.

**Comment by psfales** [comment]
FWIW,  changing the temperature with the physical buttons also pops up a message on the physical display saying "Hold until 10:00pm."  Pressing the "Hold" button converts it to a permanent hold.     Not sure how it would look if no schedule is set.

**Comment by psfales** [comment]
For the record, I'm doing everything I need with direct calls to submit_control_changes(), so thanks for exposing that interface!   The other methods would be nice, but I can workaround it for now.

**Comment by csm10495** [comment]
Closed by 81bfec5. Included in 0.1.30 via `end_hold()`, `set_temp_cool_setpoint()`, and `set_temp_heat_setpoint()`. Thanks!
